### PR TITLE
[stable/kured] feature: Add autolock to define when to execute kured

### PR DESCRIPTION
**Pull request by mtparet**
## What this PR does / why we need it:

Be able to define when to allow kured to be executed.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/kured/values.yaml [hunk 1]
```diff
@@ -16,8 +16,17 @@ serviceAccount:
   create: true
   name:
 
+autolock:
+  enabled: false
+  image:
+    repository: honestica/k8s-tools
```
Comment (mtparet) [inline stable/kured/values.yaml]: if there is a better image to use, we can switch.

**Comment by mtparet** [comment]
@plumdog @patrickmslatteryvt could you take a look at this PR ?

File: stable/kured/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: "1.2.0"
 description: A Helm chart for kured
 name: kured
-version: 1.3.1
+version: 1.3.2
```
Comment (plumdog) [inline stable/kured/Chart.yaml]: Given this adds a feature, maybe deserves a minor version increase.

File: stable/kured/templates/unlock-cronjob.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{{- if .Values.autolock.enabled }}
```
Comment (plumdog) [inline stable/kured/templates/unlock-cronjob.yaml:1]: Could make the two cronjobs use the same template (there's quite a bit of repetition between them, and they are "twins", so their sameness is inherent) which might be tidier, but might be harder to understand.

File: stable/kured/values.yaml (hunk 1 above)
Comment (plumdog) [reply stable/kured/values.yaml]: Velero uses `docker.io/bitnami/kubectl:1.14.1`, which is maybe better? (see https://github.com/helm/charts/blob/master/stable/velero/templates/cleanup-crds.yaml)

**Comment by patrickmslatteryvt** [comment]
Great idea while https://github.com/weaveworks/kured/pull/66 is in PR limbo.

**Comment by patrickmslatteryvt** [comment]
> @mtparet What is this feature and how does it work? I can't find it mentioned in https://github.com/weaveworks/kured anywhere. Is this an true feature of kured, or something that interacts with the locking ~configmap~ annotation that kured uses?

Yeah, I'd have to agree that some more explanation of how this works in the readme would greatly benefit new users.

File: stable/kured/templates/unlock-cronjob.yaml (hunk 3 above)
Comment (mtparet) [reply stable/kured/templates/unlock-cronjob.yaml:1]: Indeed but I tried but I don't know how to do that, using functions in .tpl  file storing the whole template ?

**Comment by mtparet** [comment]
@patrickmslatteryvt @plumdog  

I have added a short description of the feature and address some comments.

File: stable/kured/templates/unlock-cronjob.yaml (hunk 3 above)
Comment (plumdog) [reply stable/kured/templates/unlock-cronjob.yaml:1]: You can do this with something like:
```yaml
{{ $root := . }}
{{- range $direction := list "lock" "unlock" }}
apiVersion: batch/v1beta1
kind: CronJob
metadata:
  name: {{ template "kured.fullname" $root }}-{{ $direction }}
...
{{- if eq $direction "lock" }}
  schedule: {{ $root.Values.autolock.scheduleLock | quote }}
{{- else }}
  schedule: {{ $root.Values.autolock.scheduleUnlock | quote }}
{{- end }}
...
{{- end }}
```
But I'm inclined to say get it merged as it is, then we can worry about dry-ing up the templates later.

**Review by plumdog — APPROVED** [review]

**Comment by mtparet** [comment]
As @plumdog approved the PR, should this one be automatically merged ? @cpanato @paulczar

**Comment by mtparet** [comment]
@plumdog it seems there is /lgtm missing to get this PR merged :)

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by maxkisung** [comment]
The configuration is a little confusing.  Using the default values, does that mean kured will only run a reboots between 4am and 6am utc?

**Pull request merged**